quarkus-debezium-parent/quarkus-debezium-engine-parent/runtime/src/main/java/io/quarkus/debezium/engine/DefaultQualifiedTableNameResolver.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.quarkus.debezium.engine;
+
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.source.SourceRecord;
+
+import io.debezium.engine.RecordChangeEvent;
+
+class DefaultQualifiedTableNameResolver implements FullyQualifiedTableNameResolver {
+
+    @Override
+    public String resolve(RecordChangeEvent<SourceRecord> event) {
```

Comment:
In general this needs to be rethhought, please see `io.debezium.relational.RelationalDatabaseConnectorConfig.tableIdMapper` - it is a bit more per-database specific.

Reply:
See my other comment @kmos, can we avoid the `TableId` notion all together & rather focus on topic name?

Reply:
@Naros Focusing on `topic` name can be a solution since effectively the records here are already processed by transforms. Wondering if there can be uses cases where the users could want to filter based on the real source ant not the `topic` when this are not 1:1.

Reply:
IMHO we need both. I'd say let's use topic by default but allow user to feref to a message field or header tho override the filter. That way it would be possible to use source info block and not make it hardcoded from relational database.

Reply:
If I think to other libraries like *spring-kafka* and in particular the `@KafkaListener`, it's possible to filter/configure by:

- topic
- ~consumer group~
- ~partition~
- `RecordFilterStrategy`


I think that we can evaluate to use concept like `topic` and `RecordFilterStrategy` which is a `predicate` of `ConsumerRecord<K,V> -> boolean` and use a similar approach. What I am wondering is there any generic *identifier* that we can use for the `source` that isn't strictly related to the Kafka domain?